[Documentation] Add clarifying comment to `ClosestProof` documentation (#32)
pokt-network/smt: docs/SMT.md

- [Overview](#overview)
- [Implementation](#implementation)
  * [Inner Nodes](#inner-nodes)
  * [Extension Nodes](#extension-nodes)
  * [Leaf Nodes](#leaf-nodes)
  * [Lazy Nodes](#lazy-nodes)
  * [Lazy Loading](#lazy-loading)
  * [Visualisations](#visualisations)
    + [General Trie Structure](#general-trie-structure)
    + [Lazy Nodes](#lazy-nodes-1)
- [Paths](#paths)
  * [Visualisation](#visualisation)
- [Values](#values)
  * [Nil values](#nil-values)
- [Hashers & Digests](#hashers--digests)
- [Proofs](#proofs)
  * [Verification](#verification)
  * [Closest Proof](#closest-proof)
  * [Compression](#compression)
  * [Serialisation](#serialisation)
- [Database](#database)

>   * [Data Loss](#data-loss)

Reviewer: Are these changes intentional?

Author: These are from my linter doing its thing

sentinel leaf that has the longest common prefix and most bits in common with
the provided hash, up to the depth of the leaf found.

This method guarantees a proof of inclusion in all cases and can be verified by
using the `VerifyClosestProof` function which requires the proof and root hash
of the trie.

NB: If the hash provided to the `ClosestProof` function is known prior to the
tree being filled and closed there is the possibility of placing a leaf where
the hash will lead. If used **as intended** the hash provided should **not** be
known prior to calling the method and the tree should not be updateable after

> the fact.

Reviewer: > If the hash provided to the `ClosestProof` function is known prior to the
tree being filled and closed there is the possibility of placing a leaf where
the hash will lead.

Can you articulate what the consequences of this are?

sentinel leaf that has the longest common prefix and most bits in common with
the provided hash, up to the depth of the leaf found.

This method guarantees a proof of inclusion in all cases and can be verified by
using the `VerifyClosestProof` function which requires the proof and root hash
of the trie.

NB: If the hash provided to the `ClosestProof` function is known prior to the

> tree being filled and closed there is the possibility of placing a leaf where

Reviewer: ```suggestion
tree being filled and closed, there is the possibility of a leaf being placed an attacker where
```

> the hash will lead. If used **as intended** the hash provided should **not** be

Reviewer: ```suggestion
the hash will lead. If used **as intended**, the hash provided should **not** be
```

> known prior to calling the method and the tree should not be updateable after

Reviewer: ```suggestion
known prior to calling the method; therefore, the tree should not be updateable after
```

> the fact. If the hash were known prior to closing the tree and not inserting
> into it anymore a leaf could be inserted into the tree in such a way that it
> would always be produced from the `ClosestProof` method (as it is deterministic).

Reviewer: Not sure if this is correct but I'm still struggling with this part. Can you rephrase "procuded from", I'm not sure what that means in this context.

```suggestion
the fact. If the last hash were known by <whom?>, after the last update but prior to closing, a leaf could be inserted into the tree in such a way that it would always be produced from the `ClosestProof` method (as it is deterministic).
```

Author: So essentially the caller of the method could place a leaf that would be guareneteed to be returned by the method IFF they know they hash prior to losing access to inserting into the tree. Have tried to make this clearer while keeping it general as this *could* be used in different ways from how we use it

sentinel leaf that has the longest common prefix and most bits in common with
the provided hash, up to the depth of the leaf found.

This method guarantees a proof of inclusion in all cases and can be verified by
using the `VerifyClosestProof` function which requires the proof and root hash
of the trie.

NB: If the hash provided to the `ClosestProof` function is known prior to the
tree being filled and closed there is the possibility of placing a leaf where

> the hash will lead. If used **as intended**, the hash provided should **not** be

Reviewer: I still am not clear on "who" we should be concerned about knowing the hash and "whom" would be placing a leaf where the hash will lead and whether they're the same actor.

known prior to calling the method; therefore, the tree should not be updateable
after the fact. If the hash were known by the caller of the method, after the

> last update but prior to closing the trie a leaf could be inserted into the tree

Reviewer: Are we calling it a tree or a trie?

Author: Technically it is a trie a prefix trie I renamed it a previous PR
